day_1/parse_tree.rb

```diff
@@ -0,0 +1,85 @@
+require_relative 'parse_tag'
+
+Tag = Struct.new(:type, :classes, :id, :name, :src, :title, :text, :children, :parent)
+# :attributes that holds all html elements
+# Tag class
+# methods on tag class to retrieve that attribute
+
+class ParseTree
+	attr_accessor :html_code, :root, :tag_type
+
+  def initialize(html_code)
+  	regex = /<.*?>/
+    @tag_type = html_string.scan(regex)
+    regex = />(.*?)</
+    @text_list = html_string.scan(regex)
+    @text_list.each do |text|
+      text.each do |text_item|
+      text_item.strip!
+      end
+    end
+    # @parse = ParseTag.new(html_code)
+    @root = nil
+  end
```

Comment:
Usually, it isn't a great idea to have too much code in the initialize function, beyond the setting of instance variables to the passed parameters.

More complex functionality could be extracted to a separate method with a more intention revealing name.